core/src/main/java/org/springframework/security/converter/RsaKeyConverters.java

```diff
@@ -103,22 +115,49 @@ private RsaKeyConverters() {
 		KeyFactory keyFactory = rsaFactory();
 		return (source) -> {
 			List<String> lines = readAllLines(source);
-			Assert.isTrue(!lines.isEmpty() && lines.get(0).startsWith(X509_PEM_HEADER),
-					"Key is not in PEM-encoded X.509 format, please check that the header begins with -----"
-							+ X509_PEM_HEADER + "-----");
+			Assert.isTrue(
+					!lines.isEmpty()
+							&& (lines.get(0).startsWith(X509_PEM_HEADER) || lines.get(0).startsWith(X509_CERT_HEADER)),
+					"Key is not in PEM-encoded X.509 format or a valid X.509 certificate, please check that the header begins with -----"
```

Comment:
Even though there were dashes in the output originally, I don't think the dashes are necessary since they are included in the constants.